src/testsets.jl

```diff
@@ -187,7 +187,16 @@ function record(ts::RAITestSet, child::TestRelTestSet)
         nothing
     end
     tc = JUnitTestCase(name, counts, nothing, child.error_message, logs)
-    junit_record!(ts.junit, tc)
+
+    # A top level RAITestSet will contain a set of testsuites, so we need to wrap the
+    # testcase in a testsuite to record it
+    if Test.get_testset_depth() == 0
```

Comment:
is this sort of logic of recording a `TestCase` to a `TestSuites` something we might want to upstream @nickrobinson251? If `ReTestItems` handled this case, we could keep the logic simpler here.

Reply:
i don't feel too strongly, but i think explicitly creating the TestSuite here seems like the right thing... TestSuite is a pretty key "layer" in the JUnit XML, so i'm not sure ReTestItems should be creating it automatically from a TestCase  e.g. you need to decide what the name of the TestSuite should be, which i think it makes sense for RAITest to decide